src/main/java/com/yelp/nrtsearch/server/luceneserver/field/IndexableFieldDef.java

```diff
@@ -86,21 +86,11 @@ protected IndexableFieldDef(String name, Field requestField) {
             : requestField.getDocValuesFormat();
 
     String similarityStr = requestField.getSimilarity();
-    if (!similarityStr.isEmpty()) {
-      if (similarityStr.equalsIgnoreCase("DefaultSimilarity")) {
-        this.similarity = new ClassicSimilarity();
-      } else if (similarityStr.equalsIgnoreCase("BM25Similarity")) {
-        // TODO pass custom k1, b values so that we can call Constructor BM25Similarity(float k1,
-        // float b)
-        this.similarity = new BM25Similarity();
-      } else {
-        // TODO support CustomSimilarity
-        assert false;
-        this.similarity = null;
-      }
-    } else {
-      this.similarity = DEFAULT_SIMILARITY;
-    }
+    Map<String, Object> similarityParams =
+        Maps.transformValues(
+            requestField.getSimilarityParams().getFieldsMap(), StructValueTransformer.INSTANCE);
```

Comment:
It looks like we have a util class for this, should we use it instead ?
https://github.com/Yelp/nrtsearch/blob/master/src/main/java/com/yelp/nrtsearch/server/utils/StructValueTransformer.java#L44

Reply:
Yep, forgot I added that